### ISSUE TITLE
wandb: last reported results wrong

### ISSUE THREAD

**Issue by dreamflasher** [fastai/fastai#3068]
I am cross posting this bug report here, because after posting over in the wandb github, I realized that it rather belongs here – because the code exists here. At least currently (https://github.com/fastai/fastai/issues/2728).

**Describe the bug**
I train my model for 5 epochs, and this is the accuracy plot in wandb:
![image](https://user-images.githubusercontent.com/31695/101783119-27dd9480-3afa-11eb-8cbe-46f016e78291.png)
And this is the fastai training output:
```
epoch     train_loss  valid_loss  accuracy  b_score  ece       nll       err  time    
0         7.291822    6.423790    0.891155  0.022018     0.056384  6.423790  0.122854           05:19    
1         6.633485    6.526669    0.906586  0.019923     0.055146  6.526669  0.132975           05:17     
2         7.929371    7.037761    0.877101  0.028144     0.071046  7.037761  0.144133           05:17     
3         6.300512    6.006740    0.909066  0.017325     0.049415  6.006740  0.115119           05:17   
4         5.873019    5.817275    0.922017  0.015450     0.049534  5.817275  0.111537           05:17      
```
The wandb plot in numbers:
```
epoch 
0     0.8912
1     0.9066
2     0.8771
3     0.9091
4     0.01545
```

So something is broken in the last epoch. 
`wandb, version 0.10.11`

```
python3 -c "import fastai; import torch; import fastcore; print(torch.__version__, fastai.__version__, fastcore.__version__)"
1.7.0+cu110 2.1.8 1.3.9
```

**Comment by dreamflasher** [comment]
@borisdayma 

Here you go:

```
import wandb
from fastai.callback.wandb import WandbCallback
from fastai.vision.all import *


def label_func(f):
    return f[0].isupper()


path = untar_data(URLs.PETS)
files = get_image_files(path / "images")
dls = ImageDataLoaders.from_name_func(path, files, label_func, item_tfms=Resize(224))
wandb.init()
learn = cnn_learner(dls, resnet34, metrics=[accuracy, error_rate], cbs=[
    WandbCallback(log_model=False)])

learn.fit_one_cycle(3, slice(0.001, 0.03))

learn.validate()
```

The issue here is `learn.validate()`. This is logging bogus values to wandb.

**Comment by dreamflasher** [comment]
Improved visualization:

```
import wandb
from fastai.callback.wandb import WandbCallback
from fastai.vision.all import *


def label_func(f):
    return f[0].isupper()


path = untar_data(URLs.PETS)
files = get_image_files(path / "images")
dls = ImageDataLoaders.from_name_func(path, files, label_func, item_tfms=Resize(224))
wandb.init()
learn = cnn_learner(dls, resnet34, metrics=[accuracy, error_rate], cbs=[
    WandbCallback(log_model=False)])

learn.fit_one_cycle(3, slice(0.001, 0.03))

val_res = dict(zip(["valid_loss", "accuracy", "error_rate"], learn.validate()))
print(val_res)
wandb.log(val_res, commit=True)
```

![image](https://user-images.githubusercontent.com/31695/104199758-24566980-5428-11eb-860d-67235c5ae352.png)

run ID: er30qr43/hearty-violet-26

**Comment by borisdayma** [comment]
I'm thinking there's an issue in one of these 2 lines:
* [metrics at `after_batch`](https://github.com/fastai/fastai/blob/c3408c273b8a5bf7d90f17c04f3149e0e578e20f/fastai/callback/wandb.py#L91)
* [metrics at `after_epoch`](https://github.com/fastai/fastai/blob/c3408c273b8a5bf7d90f17c04f3149e0e578e20f/fastai/callback/wandb.py#L112)

My guess is that some of these metrics are either not calculated (and set to 0) or are named differently (maybe `loss` instead of `valid_loss`).

Based on the issue we can either tackle it in the logger or in the `validate` function.

**Comment by dreamflasher** [comment]
I think it would be nice if calling `validate` would log the correct results (rather fix the logging through `validate`, instead of deactivating it).
